torch/utils/data/datasets/batchdataset.py

```diff
@@ -55,3 +58,205 @@ def __len__(self) -> int:
+            if isinstance(shape, int):
+                shape = [shape]
+            elif isinstance(shape, Sequence):
+                if not all(isinstance(s, int) for s in shape):
+                    raise RuntimeError("The padded shape {} should be a one-dimensional "
+                                       "list/tuple of integers.".format(shape))
+            else:
+                raise RuntimeError("Invalid padded shape {}.".format(shape))
+            shape = torch.Size(shape)
+
+        if isinstance(value, Sequence) and len(value) == 1:
+            value = value[0]
+
+        new_batch = []
+        for elem in batch:
+            if elem.shape != torch.Size([]) and len(elem.shape) != len(shape):
+                raise RuntimeError("The padded shape {} should have same number of "
+                                   "dimensions as the shape of the element {}."
+                                   .format(list(shape), list(elem.shape)))
+
+            if elem.shape == shape:
+                new_batch.append(elem)
+            else:
+                v = 0 if value is None else value
+                new_elem = elem.new_full(shape, v)
+                slices = None
+                if elem.shape == torch.Size([]):
+                    slices = [slice(0, 1) for i in range(len(shape))]
+                else:
+                    slices = [slice(0, n) for n in elem.shape]
+                # Raises RuntimeError for the shape of element larger than
+                # the padded shape on any dimension
+                try:
+                    new_elem[slices] = elem
+                except RuntimeError:
+                    raise RuntimeError("Tensor can not be padded to the target shape. "
+                                       "Target: {}; Tensor: {}"
+                                       .format(list(shape), list(elem.shape)))
+                # As in-place operation can not be used for leaf variable,
+                # convert new element to requires_grad after the operation
+                if elem.requires_grad:
+                    new_elem.requires_grad_()
+                new_batch.append(new_elem)
+        return new_batch
+
+    # Recursive pad element into shape with value
+    @staticmethod
+    def _pad(batch, shape: Optional[PaddedShapeType], value: Optional[PaddedValueType]):
+        elem = batch[0]
+        elem_type = type(elem)
+        if not all(isinstance(elem, elem_type) for elem in batch):
+            raise RuntimeError("Each element in the list of batch should be same type.")
+
+        # Find largest size along each dimension
+        if shape is None:
+            shape = PaddedBatchIterableDataset._get_max_size(batch)
+
+        if isinstance(elem, torch.Tensor):
+            return PaddedBatchIterableDataset._pad_tensor(batch, shape, value)
+        # Do not apply padding to float/integer/complex number and string
+        elif isinstance(elem, (float, int, complex, str)):
+            return batch
```

Comment:
TODO: Add Numpy support
Do not apply padding for string, because text ofter requires PaddedBatch after it is tokenized to tensor or list of tensor.